src/platforms/windows/windowsappimageprovider.cpp

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "logger.h"
+#include "leakdetector.h"
+#include "windowsappimageprovider.h"
+#include <QObject>
+#include <QQuickImageProvider>
+#include <QtWin>
+
+#include <QCommandLineParser>
+#include <QCommandLineOption>
+#include <QDir>
+#include <QFileInfo>
+#include <QGuiApplication>
+#include <QImage>
+#include <QPixmap>
+#include <QScopedArrayPointer>
+#include <QStringList>
+#include <QSysInfo>
+
+#include <iostream>
+
+#include <shellapi.h>
+#include <comdef.h>
+#include <commctrl.h>
+#include <objbase.h>
+#include <commoncontrols.h>
+
+
+namespace {
+Logger logger(LOG_WINDOWS, "WindowsAppImageProvider");
+}
+
+
+WindowsAppImageProvider::WindowsAppImageProvider(QObject* parent)
+    : QQuickImageProvider(QQuickImageProvider::Pixmap), QObject(parent) {
+  MVPN_COUNT_CTOR(WindowsAppImageProvider);
+}
+
+WindowsAppImageProvider::~WindowsAppImageProvider() {
+  MVPN_COUNT_DTOR(WindowsAppImageProvider);
+}
+
+
+QPixmap WindowsAppImageProvider::requestPixmap(const QString& path, QSize* size,
+                                      const QSize& requestedSize){
+    Q_UNUSED(requestedSize);
+    logger.log() << "Request Image for " << path;
+
+    const QString nativePath = QDir::toNativeSeparators(path);
+    const auto *sourceFileC = reinterpret_cast<const wchar_t *>(nativePath.utf16());
+    const auto iconCount = ExtractIconEx(sourceFileC, -1, nullptr, nullptr, 0);
```

Comment:
is it an DWORD? or an int? For Windows calls, maybe `auto` is not the right choice for readability.